src/client/websocket/WebSocketShard.js

```diff
@@ -475,12 +475,22 @@ class WebSocketShard extends EventEmitter {
       return;
     }
     const hasGuildsIntent = new Intents(this.manager.client.options.intents).has(Intents.FLAGS.GUILDS);
-    // Step 2. Create a 15s timeout that will mark the shard as ready if there are still unavailable guilds
+    // Step 2. Create a timeout that will mark the shard as ready if there are still unavailable guilds
+    // * the timeout should be 15 seconds by default, but this can be optionally be changed in the client options
+
+    var guildTimeoutAmount = 15000;
+
+    if (this.manager.client.options.fetchGuildTimeout) {
+      guildTimeoutAmount = this.manager.client.options.fetchGuildTimeout;
+    }
+
+    var guildTimeoutAmountInSeconds = guildTimeoutAmount / 1000;
+
     this.readyTimeout = setTimeout(
       () => {
         this.debug(
           `Shard ${hasGuildsIntent ? 'did' : 'will'} not receive any more guild packets` +
-            `${hasGuildsIntent ? ' in 15 seconds' : ''}.\n   Unavailable guild count: ${this.expectedGuilds.size}`,
+            `${hasGuildsIntent ? `in ${guildTimeoutAmountInSeconds} seconds` : ''}.\n   Unavailable guild count: ${this.expectedGuilds.size}`,
```

Comment:
```suggestion
            `${hasGuildsIntent ? `in ${Math.round(guildTimeoutAmount / 1000)} seconds` : ''}.\n   Unavailable guild count: ${this.expectedGuilds.size}`,
```

Reply:
What if we just used ms?

Reply:
Well, it's the same. I personally think that if the option is in ms maybe the message logged should be in ms too but it's better that official djs collaborators decide lol

Reply:
The option should probably be in seconds instead since you're not gonna want nor need a timeout of less than 1 second

Reply:
I think it's kinda weird to have decimals in the options but I'm not entirely sure. I think the console log and options should be the same formatting (in milliseconds) to make the experience smoother for devs.

Reply:
But you shouldn’t need decimals for the same reason you shouldn’t need milliseconds. Your guilds will never arrive that fast so it’s unreasonable to set a lower timeout. If the dev REALLY wants that then they can use decimals, but I don’t see why they would

Reply:
Thank you for pointing that out. I was looking at the client options and the request-based ones all use milliseconds. Devs are already quite familiar with setTimeout in JS and other similar stuff being based on milliseconds.

Does this change have to be fulfilled for the option to be merged into the repo?

Reply:
Yeah that’s true. I’m not a maintainer I’m just giving my personal opinion so let’s wait for one of the @discordjs/the-big-3 to say something about this